pkg/apis/machine/v1alpha1/machineclass_types.go

```diff
@@ -58,3 +61,13 @@ type MachineClassList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []MachineClass `json:"items"`
 }
+
+// ExpectedNodeDetails contains subfields the track all scale from zero resources
+type ExpectedNodeDetails struct {
+	CPU          int64  `json:"cpu"`
+	GPU          int64  `json:"gpu"`
+	InstanceType string `json:"instanceType"`
+	Memory       int64  `json:"memory"`
+	Region       string `json:"region"`
+	Zone         string `json:"zone"`
```

Comment:
We need to make them optional fields with comments and pointers. These changes have to also be added in the actual API files.

Reply:
Sure, Prashanth. I will make these changes.